source/extensions/io_socket/user_space/io_handle_impl.h

```diff
@@ -88,6 +88,7 @@ class IoHandleImpl final : public Network::IoHandle,
 
   Api::SysCallIntResult shutdown(int how) override;
   absl::optional<std::chrono::milliseconds> lastRoundTripTime() override { return absl::nullopt; }
+  absl::optional<std::string> interfaceName() override { return absl::nullopt; }
```

Comment:
@florincoras @rojkov I believe the same implementation that I wrote for the IoSocketHandleImpl would work here. Do you think it would be worth it to move the implementation to the virtual class? Given that the implementation is based on `localAddress()` and the os syscalls available it doesn't seem like it would necessarily need to be outside of the virtual class.

Reply:
The virtual class is supposed to be just an interface enforcing concrete implementations at compile-time like this trivial stub. Otherwise there's a chance of calling a suboptimal implementation (which e.g. checks os syscall availability first). And in this particular case it could be even an error if something called a user space handle's `interfaceName()`.

But I agree the implementation should be shared somehow where it makes sense. In #19082 I'll need the same code. Even more to that - I need the same implementation for `localAddress()` and `peerAddress()`. Maybe utility functions would suffice?

Reply:
Agreed with @rojkov. It won't also work for vcl because the interfaces are in another user space process (vpp), so not available via syscalls. 

The idea might not be too bright but one could argue that there should've been a syscall to get a socket's `localAddress()`, `peerAddress()` and the interfaceName for a `localAddress()`, so one option would be to add these utility functions to syscall interfaces. The downside however is that we'll pollute those with non-posix apis.

Reply:
@rojkov @florincoras thanks for your thoughtful responses. It makes sense to me. I also realize how little experience I have of this line of work. If you dont think that the current addition to the interface is egregious, do you all mind if I implement here as is and let you all handle this when/if you tackle moving localAddress() and peerAddress around?

Reply:
I don't, I'm fine with it as it is for now.